inst/templates/ad_adpc.R

```diff
@@ -0,0 +1,513 @@
+  # Derive event ID and nominal relative time from first dose (NFRLT)
+  mutate(
+    EVID = 0,
+    DRUG = PCTEST,
+    NFRLT = if_else(PCTPTNUM < 0, 0, PCTPTNUM), .after = USUBJID
+  )
+
+
+# ---- Get dosing information ----
+
+ex <- ex %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = vars(STUDYID, USUBJID)
+  ) %>%
+  # Keep records with nonzero dose
+  filter(EXDOSE > 0) %>%
+  # Add time and set missing end date to start date
+  # Impute missing time to 00:00:00
+  # Derive Analysis Start and End Dates
+  derive_vars_dtm(
+    new_vars_prefix = "AST",
+    dtc = EXSTDTC,
+    date_imputation = "FIRST",
+    time_imputation = "00:00:00",
+    ignore_seconds_flag = FALSE,
+    flag_imputation = "auto"
+  ) %>%
+  derive_vars_dtm(
+    new_vars_prefix = "AEN",
+    dtc = EXENDTC,
+    date_imputation = "LAST",
+    time_imputation = "00:00:00",
+    ignore_seconds_flag = FALSE,
+    flag_imputation = "auto"
+  ) %>%
+  # Derive event ID and nominal relative time from first dose (NFRLT)
+  mutate(
+    EVID = 1,
+    NFRLT = 24 * (VISITDY - 1), .after = USUBJID
+  ) %>%
+  # Set missing end dates to start date
+  mutate(AENDTM = case_when(
+    is.na(AENDTM) ~ ASTDTM,
+    TRUE ~ AENDTM
+  )) %>%
+  # Derive dates from date/times
+  derive_vars_dtm_to_dt(vars(ASTDTM)) %>%
+  derive_vars_dtm_to_dt(vars(AENDTM))
+
+
+# ---- Expand dosing records between start and end dates ----
+# Updated function includes nominal_time parameter
+
+ex_exp <- ex %>%
+  create_single_dose_dataset(
+    dose_freq = EXDOSFRQ,
+    start_date = ASTDT,
+    start_datetime = ASTDTM,
+    end_date = AENDT,
+    end_datetime = AENDTM,
+    nominal_time = NFRLT,
+    lookup_table = dose_freq_lookup,
+    lookup_column = CDISC_VALUE,
+    keep_source_vars = vars(
+      STUDYID, USUBJID, EVID, EXDOSFRQ, EXDOSFRM,
+      NFRLT, EXDOSE, EXTRT, ASTDT, ASTDTM, AENDT, AENDTM,
+      VISIT, VISITNUM, VISITDY,
+      TRT01A, TRT01P, DOMAIN, EXSEQ, !!!adsl_vars
+    )
+  ) %>%
+  # Derive AVISIT based on nominal relative time
+  mutate(
+    AVISITN = NFRLT %/% 24 + 1,
+    AVISIT = paste("Day", AVISITN),
+    ADTM = ASTDTM,
+    DRUG = EXTRT
+  ) %>%
+  # Derive dates and times from datetimes
+  derive_vars_dtm_to_dt(vars(ADTM)) %>%
+  derive_vars_dtm_to_tm(vars(ADTM)) %>%
+  derive_vars_dtm_to_tm(vars(ASTDTM)) %>%
+  derive_vars_dtm_to_tm(vars(AENDTM)) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = vars(ADT))
+
+
+# ---- Find first dose per treatment per subject ----
+# ---- Join with ADPC data and keep only subjects with dosing ----
+
+adpc <- adpc %>%
+  derive_vars_merged(
+    dataset_add = ex_exp,
+    filter_add = (EXDOSE > 0 & !is.na(ADTM)),
+    new_vars = vars(FANLDTM = ADTM),
+    order = vars(ADTM, EXSEQ),
+    mode = "first",
+    by_vars = vars(STUDYID, USUBJID, DRUG)
+  ) %>%
+  filter(!is.na(FANLDTM)) %>%
+  # Derive AVISIT from nominal relative time
+  mutate(
+    AVISITN = NFRLT %/% 24 + 1,
```

Comment:
ohh can we explain what this is doing here?!?

Reply:
Yes, added additional comments to clarify.